### PR TITLE
Reconfig logger format to follow kyma community concept

### PR DESCRIPTION
**Pull request by ruanxin**
Implement based on kyma community concept - Consistent Logging
https://github.com/kyma-project/community/tree/main/concepts/observability-consistent-logging

### CODE REVIEW

File: operator/main.go [hunk 1]
```diff
@@ -124,6 +119,17 @@ func main() {
 	setupManager(flagVar, controllers.NewCacheFunc(), scheme)
 }
 
+func configLogger() logr.Logger {
```
Comment (ruanxin) [inline operator/main.go:122]: based on this issue: https://github.com/uber-go/zap/issues/485#issuecomment-359662417

File: operator/main.go [hunk 2]
```diff
@@ -124,6 +119,17 @@ func main() {
...
+	atomicLevel := zap.NewAtomicLevel()
+	atomicLevel.SetLevel(zap.DebugLevel)
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.TimeKey = "timestamp"
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
```
Comment (jakobmoellersap) [inline operator/main.go:127]: Why not use RFC3339?

File: operator/main.go (hunk 1 above)
Comment (jakobmoellersap) [inline operator/main.go:122]: Can we verify what the correct logger settings are for CLS / Logging Service with Monitoring Team? I think we should then use these settings and apply them globally

File: operator/main.go (hunk 2 above)
Comment (ruanxin) [reply operator/main.go:127]: RFC3339     = "2006-01-02T15:04:05Z07:00"
RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
	
The standard RFC3339 only precise to second, we can use RFC3339Nano

File: operator/main.go (hunk 1 above)
Comment (ruanxin) [reply operator/main.go:122]: Yes, we have one https://github.com/kyma-project/community/tree/main/concepts/observability-consistent-logging, I'll have a check and do changes accordingly.

File: operator/main.go (hunk 2 above)
Comment (jakobmoellersap) [reply operator/main.go:127]: RFC Nano is implementation and os-dependant and vary with precision. Also I do not think we need nano level precision on controllers.

File: operator/controllers/kyma_controller.go [hunk 3]
```diff
@@ -79,7 +77,7 @@ type KymaReconciler struct {
 
 func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
-	logger.Info("reconciling modules", "resource", req.NamespacedName.String())
```
Comment (ruanxin) [inline operator/controllers/kyma_controller.go:82]: The controller runtime logs already contains following information, don't need to duplicate mention resource.
```
"context":{"controller":"kyma","controllerGroup":"operator.kyma-project.io","controllerKind":"Kyma","Kyma":{"name":"kyma-xin-1-0","namespace":"default"},"namespace":"default","name":"kyma-xin-1-0","reconcileID":"5322de8a-ebfa-4c3a-b5c0-2ac751f80165"}
```

File: operator/main.go (hunk 1 above)
Comment (ruanxin) [reply operator/main.go:122]: with currently refactoring, one example log like below:
```
{
    "level": "INFO",
    "date": "2022-11-03T15:01:27.294739+01:00",
    "caller": "controllers/kyma_controller.go:80",
    "msg": "reconciling modules",
    "context": {
        "controller": "kyma",
        "controllerGroup": "operator.kyma-project.io",
        "controllerKind": "Kyma",
        "Kyma": {
            "name": "kyma-xin-1-0",
            "namespace": "default"
        },
        "namespace": "default",
        "name": "kyma-xin-1-0",
        "reconcileID": "5322de8a-ebfa-4c3a-b5c0-2ac751f80165"
    }
}
```

File: operator/main.go (hunk 2 above)
Comment (ruanxin) [reply operator/main.go]: I think at least we need millisecond level, first is the reconciliation is really fast, I think it's better to have distinguish for each reconciliation in time level, especially when the situation happens when we have many kyma and concurrent workers configured. btw, this is the default timestamp of the runtime controller after convert with k9s, it's by default use nana level timestamp. within 1 second, we currently have hundreds of log messages in load test cluster.

Regarding os-dependant, which operation system not support it? But the operator running inside k8s container with linux core, should be fine, or?
```
│ 2022-11-03T14:22:38.510444813Z manager 1.6674853585103135e+09    INFO    reconciling modules    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--34-171","namespace":"default"}, "namespace": "default", "name": "kyma--34-171", │
│ 2022-11-03T14:22:38.510478085Z manager 1.6674853585103347e+09    INFO    reconciling modules    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--42-26","namespace":"default"}, "namespace": "default", "name": "kyma--42-26", " │
│ 2022-11-03T14:22:38.511603205Z manager 1.6674853585114877e+09    INFO    reconciling modules    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--18-100","namespace":"default"}, "namespace": "default", "name": "kyma--18-100", │
│ 2022-11-03T14:22:38.511660618Z manager 1.6674853585115814e+09    INFO    reconciling modules    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--33-4","namespace":"default"}, "namespace": "default", "name": "kyma--33-4", "re │
│ 2022-11-03T14:22:38.526287964Z manager 1.6674853585261514e+09    INFO    reconciling modules    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--4-186","namespace":"default"}, "namespace": "default", "name": "kyma--4-186", " │
│ 2022-11-03T14:22:38.551514769Z manager 1.6674853585513988e+09    INFO    reconciling modules    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--45-51","namespace":"default"}, "namespace": "default", "name": "kyma--45-51", " │
│ 2022-11-03T14:22:38.561130942Z manager 1.667485358561005e+09    INFO    syncing state    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--18-100","namespace":"default"}, "namespace": "default", "name": "kyma--18-100", "recon │
│ 2022-11-03T14:22:38.564486954Z manager 1.6674853585643847e+09    INFO    syncing state    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--33-4","namespace":"default"}, "namespace": "default", "name": "kyma--33-4", "reconcil │
│ 2022-11-03T14:22:38.565699159Z manager 1.6674853585655954e+09    INFO    syncing state    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--34-171","namespace":"default"}, "namespace": "default", "name": "kyma--34-171", "reco │
│ 2022-11-03T14:22:38.568483257Z manager 1.6674853585683784e+09    INFO    syncing state    {"controller": "kyma", "controllerGroup": "operator.kyma-project.io", "controllerKind": "Kyma", "Kyma": {"name":"kyma--42-26","namespace":"default"}, "namespace": "default", "name": "kyma--42-26", "reconc │

File: operator/main.go [hunk 4]
```diff
@@ -124,6 +119,20 @@ func main() {
 	setupManager(flagVar, controllers.NewCacheFunc(), scheme)
 }
 
+func configLogger() logr.Logger {
+	// The following settings is based on kyma community Improvement of log messages usability
+	// https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md#log-structure
+	atomicLevel := zap.NewAtomicLevel()
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.TimeKey = "date"
+	encoderConfig.EncodeTime = zapcore.RFC3339NanoTimeEncoder
```
Comment (jakobmoellersap) [inline operator/main.go:128]:
```suggestion
	encoderConfig.EncodeTime = zapcore.RFC3339TimeEncoder
```

**Comment by jakobmoellersap** [comment]
Fixes #299

**Review by jakobmoellersap — COMMENTED** [review]
Lint failing, please double check

File: operator/main.go [hunk 5]
```diff
@@ -124,6 +119,20 @@ func main() {
 	setupManager(flagVar, controllers.NewCacheFunc(), scheme)
 }
 
+func configLogger() logr.Logger {
+	// The following settings is based on kyma community Improvement of log messages usability
+	// https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md#log-structure
+	atomicLevel := zap.NewAtomicLevel()
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.TimeKey = "date"
+	encoderConfig.EncodeTime = zapcore.RFC3339TimeEncoder
```
Comment (ruanxin) [inline operator/main.go:128]:
```suggestion
	encoderConfig.EncodeTime = zapcore.RFC3339NanoTimeEncoder
```

**Pull request merged**